internal/querynode/shard_cluster.go

```diff
@@ -381,20 +400,166 @@ func (sc *ShardCluster) getSegment(segmentID int64) (*shardSegmentInfo, bool) {
 }
 
 // segmentAllocations returns node to segments mappings.
+// calling this function also increases the reference count of related segments.
 func (sc *ShardCluster) segmentAllocations(partitionIDs []int64) map[int64][]int64 {
 	result := make(map[int64][]int64) // nodeID => segmentIDs
-	sc.mut.RLock()
-	defer sc.mut.RUnlock()
+	sc.mut.Lock()
```

Comment:
Question: why change to mutex lock if we only read guarded variables?

Reply:
`segmentAllocations` actually change the `inUse` reference count now.

Reply:
got it